tests/compilation_cache_test.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl.testing import absltest
+import hashlib
+from jax.experimental.compilation_cache import compilation_cache as cc
+import jax
+import jax.test_util as jtu
+import numpy as np
+
+class CompilationCacheTest(jtu.JaxTestCase):
+
+  def test_cache_key_type(self):
+    computation = jax.xla_computation(lambda x, y: x + y)(1, 1)
+    compile_options = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    self.assertEqual(type(cc.get_cache_key(computation, compile_options)), str)
+
+  def test_compile_options(self):
+    compile_options_not_filled = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    compile_options_filled = self.fill_compile_options()
+    hash_obj1 = self.create_sudo_hash_obj(cc._hash_compile_options, compile_options_filled)
+    hash_obj2 = self.create_sudo_hash_obj(cc._hash_compile_options, compile_options_filled)
+    hash_obj3 = self.create_sudo_hash_obj(cc._hash_compile_options, compile_options_not_filled)
+    self.assertEqual(hash_obj1.hex(), hash_obj2.hex())
+    self.assertNotEqual(hash_obj1.hex(), hash_obj3.hex())
+
+  def test_executable_build_options(self):
+    compile_options_not_filled = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    compile_options_filled = self.fill_compile_options()
+    hash_obj1 = self.create_sudo_hash_obj(cc._hash_executable_build_options, compile_options_filled.executable_build_options)
```

Comment:
Add a line break to make this line shorter (here and similarly-long lines below)